velox/functions/prestosql/Repeat.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+// See documentation at https://prestodb.io/docs/current/functions/array.html
+class RepeatFunction : public exec::VectorFunction {
+ public:
+  static constexpr int32_t kMaxResultEntries = 10000;
+
+  bool isDefaultNullBehavior() const override {
+    // For repeat(null, count), it should output an array with
+    // repeating null for "count" times.
+    return false;
+  }
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    const auto numRows = rows.size();
+    auto pool = context.pool();
+    auto count = 0;
+    DecodedVector* countDecoded = nullptr;
+
+    if (args[1]->isConstantEncoding()) {
+      if (args[1]->as<ConstantVector<int32_t>>()->isNullAt(0)) {
+        // If count is a null constant, the result should be all nulls.
+        context.moveOrCopyResult(
+            BaseVector::createNullConstant(outputType, numRows, pool),
+            rows,
+            result);
+        return;
+      }
+      count = args[1]->as<ConstantVector<int32_t>>()->valueAt(0);
+      checkCount(count);
+    } else {
+      exec::DecodedArgs decodedArgs(rows, args, context);
+      countDecoded = decodedArgs.at(1);
+    }
+
+    // Calculate the total number of the new elements.
+    auto totalCount = 0;
+    if (args[1]->isConstantEncoding()) {
+      totalCount = count * numRows;
+    } else {
+      rows.applyToSelected([&](vector_size_t row) {
+        auto count = countDecoded->isNullAt(row)
+            ? 0
+            : countDecoded->valueAt<int32_t>(row);
+        checkCount(count);
```

Comment:
Errors encountered on some rows should not prevent computation for other rows when expression run inside TRY:

https://facebookincubator.github.io/velox/develop/scalar-functions.html#try-expression-support

See #3527 for a similar bug fix.

Reply:
Don't we want to fail the whole query if any row exceeds the threshold to be consistent with the current Presto's behavior?
![image](https://user-images.githubusercontent.com/4384058/210476739-6ae7f5d7-2c65-4dd7-ab1a-28d9e8fb1fe1.png)

Reply:
@zacw7 The query should not fail if repeat call is wrapped in a try. (Next time, share text snippets instead of screenshots whenever possible. Screenshots are not accessible to everybody and do not allow for copy-paste.)

```
presto> select repeat(null, -1);
Query 20230104_124903_45991_ae7g6 failed: count argument of repeat function must be greater than or equal to 0

presto> select try(repeat(null, -1));
 _col0
-------
 NULL
(1 row)

Query 20230104_124909_46008_ae7g6, FINISHED, 1 node
Splits: 5 total, 5 done (100.00%)
259ms [0 rows, 0B] [0 rows/s, 0B/s]
```